packages/aws-cdk-lib/aws-route53/lib/key-signing-key.ts

```diff
@@ -0,0 +1,184 @@
+   *
+   * The KMS Key must be unique for each KSK within a hosted zone. Additionally, the
+   * KMS key must be an asymetric customer-managed key using the ECC_NIST_P256 algorithm.
+   *
+   * @see https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/dns-configuring-dnssec-cmk-requirements.html
+   */
+  readonly kmsKey: kms.IKey;
+
+  /**
+   * The name for the key signing key.
+   *
+   * This name must be unique within a hosted zone.
+   *
+   * @default an autogenerated name
+   */
+  readonly keySigningKeyName?: string;
+
+  /**
+   * The status of the key signing key.
+   *
+   * @default ACTIVE
+   */
+  readonly status?: KeySigningKeyStatus;
+}
+
+/**
+ * The status for a Key Signing Key.
+ */
+export enum KeySigningKeyStatus {
+  /** The KSK is being used for signing. */
+  ACTIVE = 'ACTIVE',
+  /** The KSK is not being used for signing. */
+  INACTIVE = 'INACTIVE',
+}
+
+/**
+ * A Key Signing Key for a Route 53 Hosted Zone.
+ */
+export interface IKeySigningKey extends IResource {
+  /**
+   * The hosted zone that the key signing key signs.
+   *
+   * @attribute
+   */
+  readonly hostedZone: IHostedZone;
+
+  /**
+   * The name of the key signing key.
+   *
+   * @attribute
+   */
+  readonly keySigningKeyName: string;
+
+  /**
+   * The ID of the key signing key, derived from the hosted zone ID and its name.
+   *
+   * @attribute
+   */
+  readonly keySigningKeyId: string;
+}
+
+/**
+ * The attributes of a key signing key.
+ */
+export interface KeySigningKeyAttributes {
+  /**
+   * The hosted zone that the key signing key signs.
+   *
+   * @attribute
+   */
+  readonly hostedZone: IHostedZone;
+
+  /**
+   * The name of the key signing key.
+   *
+   * @attribute
+   */
+  readonly keySigningKeyName: string;
+}
+
+/**
+ * A Key Signing Key for a Route 53 Hosted Zone.
+ *
+ * @resource AWS::Route53::KeySigningKey
+ */
+export class KeySigningKey extends Resource implements IKeySigningKey {
+
+  /**
+   * Imports a key signing key from its attributes.
+   */
+  public static fromKeySigningKeyAttributes(scope: Construct, id: string, attrs: KeySigningKeyAttributes): IKeySigningKey {
+    class Import extends Resource implements IKeySigningKey {
+      public readonly keySigningKeyName: string;
+      public readonly hostedZone: IHostedZone;
+
+      constructor() {
+        super(scope, id);
+        this.keySigningKeyName = attrs.keySigningKeyName;
+        this.hostedZone = attrs.hostedZone;
+      }
+
+      get keySigningKeyId() {
+        return `${this.hostedZone.hostedZoneId}|${this.keySigningKeyName}`;
+      }
+    }
+
+    return new Import();
+  }
+
+  public readonly hostedZone: IHostedZone;
+  public readonly keySigningKeyName: string;
+  public readonly keySigningKeyId: string;
+
+  constructor(scope: Construct, id: string, props: KeySigningKeyProps) {
+    super(scope, id, {
+      physicalName: props.keySigningKeyName ?? Lazy.string({
```

Comment:
We should add validation for `keySigningKeyName` according to the [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-route53-keysigningkey.html#cfn-route53-keysigningkey-name).

Reply:
I don't actually love doing this because if the service team changes this, we have to make an additional code change to allow customers to use the new pattern as well. I tend to think that the deploy time validation should be good enough in this case.

Another option I've been considering is trying to add something into our overall validation at synth that takes in any of these patterns and checks against it, but that's much much larger project than this.